appcli/variables_manager.py

```diff
@@ -95,34 +98,119 @@ def set_all_variables(self, variables: Dict):
+
         """
-        try:
-            raw_data = self.configuration_file.read_text(encoding="utf-8")
+        config_variables_main = self.__get_configuration_main()
+        config_variables = self.__get_configuration_extra(config_variables_main)
+        return config_variables_main | config_variables
 
-            # If the file is empty, the YAML library will load as `None`. Since
-            # we expect this function to return a valid dict, we return an
-            # empty dictionary if it's empty.
-            yaml_data = self.yaml.load(raw_data)
-            if yaml_data is None:
-                return {}
-            return yaml_data
+    def __get_configuration_main(self) -> Dict:
+        """Gets the configuration from the main `settings.yml` file.
 
+        Returns:
+            Dict: The configuration data from the main configuration file.
+        """
+        config_variables_main = dict()
+        try:
+            data_string = self.__read_configuration_source(self.configuration_file)
+            config_variables_main |= self.__convert_yaml_to_dict(
+                data_string, self.configuration_file.stem
+            )
         except Exception as ex:
             raise Exception(
-                f"Could not read configuration file at [{self.configuration_file}]"
+                f"Could not read main configuration file at [{self.configuration_file}]"
             ) from ex
+        return config_variables_main
+
+    def __get_configuration_extra(self, variables: Dict) -> Dict:
+        """Gets the configuration from the additional configuration files.
+
+        Args:
+            variables (Dict): Variables used to populate the template(s).
+
+        Returns:
+            Dict: The configuration data from the additional configuration files.
+                Each config file is a seperate dictionary with its filename as the key.
+        """
+        config_variables = dict()
+        for config_file in self.extra_configuration_files.glob("*"):
+            try:
+                data_string = self.__read_configuration_source(config_file)
+                if config_file.name.endswith(".j2"):  # Jinja2 file.
+                    data_string = self.__convert_jinja_to_yaml(data_string, variables)
+                config_variables |= self.__convert_yaml_to_dict(
+                    data_string, config_file.stem
+                )
+            except Exception as ex:
+                raise Exception(
+                    f"Could not read configuration file at [{config_file}]"
+                ) from ex
+        return config_variables
 
     def __save(self, variables: Dict):
         """Saves the supplied Dict of variables to the configuration file
 
         Args:
-            variables (Dict): the variables to save
+            variables (Dict): All the variables to save (main config dictionary will be extracted).
         """
+        variables = variables[self.configuration_file.stem]
```

Comment:
The aim of this function is to take in the dict of 'main' settings, and save it to the 'main' settings file. This change complicates the way that this happens: it now takes in all the merged settings (main + extra), and does some manipulation to pull out the main settings, and save it to the 'main' settings file.

This change heavily couples the idea of the main and extra settings files being merged together in logic, but separated in files.

We should always aim to treat our 'main' settings and 'extra' settings as separate as possible. We only need the VariablesManager to merge the 'main' and 'extra' settings together for templating purposes.

Reply:
Fixed.